PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt

```diff
@@ -93,8 +99,20 @@ class PayPalClient internal constructor(
             error: Exception? ->
             if (payPalResponse != null) {
                 payPalContextId = payPalResponse.pairingId
+                linkType = if (internalPayPalClient.isAppSwitchEnabled(payPalRequest) &&
+                    internalPayPalClient.isPayPalInstalled(context)) {
+                    LinkType.UNIVERSAL
+                } else {
+                    LinkType.DEEPLINK
+                }
+
                 try {
                     payPalResponse.browserSwitchOptions = buildBrowserSwitchOptions(payPalResponse)
+
+                    if (internalPayPalClient.isAppSwitchEnabled(payPalRequest)) {
+                        braintreeClient.sendAnalyticsEvent(PayPalAnalytics.APP_SWITCH_STARTED, analyticsParams)
```

Comment:
Should this analytic event be moved up to right after the `TOKENIZE_STARTED` event? I assume we want to trigger this immediately at the start of the flow

Reply:
I think we want to trigger this event right before we launch the Venice app

Reply:
Exactly, this event needs to be called before launching the Venice app. The callback that comes afterward (`callback.onPayPalPaymentAuthRequest`) is the one that tells the fragment to launch the Venice app